content/articles/recover-deleted-files-with-foremost/index.md

```diff
@@ -0,0 +1,121 @@
+The command gives the following output.
+
+![start recovery](/engineering-education/recover-deleted-files-with-foremost/recover.png)
+
+After the recovery process has run to completion, the recovered files will be saved in the output file path. You may need to practice some patience as the process may take longer than usual depending on the drive's size. Below is the output file.
+
+![Output](/engineering-education/recover-deleted-files-with-foremost/output.png)
+
+### 2. Recovering files from a hard disk
+
+To list the drives attached to your machine, use the command below:
+
+```bash
+sudo fdisk -l
+```
+
+The above command will list the drives attached to your computer and all their partitions. The diagram below shows its output.
+
+![List Drives](/engineering-education/recover-deleted-files-with-foremost/drive.png)
+
+**Recovery of files from a hard drive can be done in two ways:**
+
+- Recover files from the entire hard drive i.e., **foremost** will recover files from all partitions of a hard drive. This process will take more time.
+- Recover files from a specific partition i.e., you can specify the partition from which **foremost** will recover the files. This is applicable where you know from which partition your deleted files were located.
+
+**To recover files from the entire hard drive use the following command:**
+
+```bash
+sudo foremost -v -t png,mp4,pdf  -i /dev/sda -o /home/Desktop/recover -T
+```
+
+Replace the hard drive path with your own. You will find the path in the output of the command used to list drives. Below is an output of the command above.
+
+![Start recovery](/engineering-education/recover-deleted-files-with-foremost/disk.png)
+
+Be patient as this process will take a while.
+
+**To recover files from a single partition, use the following command:**
+
+```bash
+sudo foremost -v -t png,mp4,pdf -i /dev/sda1  -o /home/Desktop/recover -T
+```
+
+Replace the partition path with your own. This process will take a while depending on the size of the partition. For larger partitions, be patient as this process may take some time.
+
+### Recovering files of all types
+
+To recover all the file types that foremost supports, use the following command.
+
+```bash
+sudo foremost -v  -q -i /dev/sdc 
+```
+
+Replace the drive path above with the one you would like to recover files from.
+
+The above command will automatically generate an output folder called output. The location of the folder will be in the home directory. The output folder is shown in the picture below.
+
+![Output](/engineering-education/recover-deleted-files-with-foremost/all.png)
+
+### Conclusion
+
+We have discussed the numerous ways you can use **foremost** to recover your deleted files. There are no limitations to what you can do with **foremost**. Use the tools we discussed above to your preference.
```

Comment:
I'm not satisfied with your conclusion statement. Write your final point clearly, providing closure to the reader, leaving them with a strong impression of the significance of using foremost within a broader context. Build up on what you've already written.